### PR TITLE
Specs views n+1 problem

### PR DESCRIPTION
**Pull request by kessio**
Hello 👋🏾 
**Project requirements**
n+1
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
Do not use the bullet gem in this exercise 

integration specs
Use Capybara to write integration tests for each view in your project.

- [x] User index page:
- [x] user show page:
- [x] User post index page
- [x] Post show page:

### CODE REVIEW

File: spec/views/users/users_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  it 'shows  number of posts' do
+    expect(page).to have_content(@user.posts_counter)
+  end
+
+  it 'shows last three posts of user' do
+    expect(page).to have_content(@post1.text)
+    expect(page).to have_content(@post2.text)
+    expect(page).to have_content(@post3.text)
+  end
```
Comment (Whoistolu) [inline spec/views/users/users_show_spec.rb]: - [ ] [REQUIRED] This test fails: Tests that assert you can see the user's first 3 posts. However, line 52 of your code describes it as 'last three posts' even though you created new posts using the Post.new() method. Kindly fix these errors.
Screenshot 2023-02-09 at 10 22 45

File: spec/views/users/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  it 'shows button for seeing all posts' do
+    expect(page).to have_link('See all posts')
+  end
+
+  it 'redirects to post show page upon click' do
+    click_link('See all posts')
+    expect(page).to have_current_path(user_posts_path(@user))
+  end
```
Comment (Whoistolu) [inline spec/views/users/users_show_spec.rb:70]: - [ ] [REQUIRED] Kindly also fix this error also as it looks like the right link can not be found.
Screenshot 2023-02-09 at 10 22 45

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @kessio 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi @kessio 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Well done implementing the requested changes 

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**